spec/policies/task_policy_spec.rb

```diff
@@ -27,26 +28,22 @@
   end
 
   context 'with a user' do
-    let(:user) { create(:user) }
+    let(:user) { create(:user, openai_api_key:) }
     let(:generic_user_permissions) { %i[index new import_start import_confirm import_uuid_check import_external] }
 
     it { is_expected.to permit_only_actions(generic_user_permissions) }
 
     context 'when user is admin' do
-      let(:user) { create(:admin) }
+      let(:user) { create(:admin, openai_api_key:) }
 
       context 'without gpt access token' do
+        let(:openai_api_key) { nil }
+
         it { is_expected.to forbid_only_actions %i[generate_test] }
       end
```

Comment:
You could just do this, because the `openai_api_key` is already set to `nil` in line 12

```suggestion
        it { is_expected.to forbid_only_actions %i[generate_test] }
```

Reply:
I will apply the same in other similar contexts